src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -744,14 +744,31 @@ internal static bool IsPackageReferenceProject(Project project)
                     if (matchingPackages.Any())
                     {
                         var topLevelPackage = matchingPackages.Single();
-                        InstalledPackageReference installedPackage;
+                        InstalledPackageReference installedPackage = default;
 
                         //If the package is not auto-referenced, get the version from the project file. Otherwise fall back on the assets file
                         if (!topLevelPackage.AutoReferenced)
                         {
                             try
                             { // In case proj and assets file are not in sync and some refs were deleted
-                                installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                if (assetsFile.PackageSpec.RestoreMetadata.CentralPackageVersionsEnabled)
+                                {
+                                    foreach (KeyValuePair<string, CentralPackageVersion> packageCentralVersion in tfmInformation.CentralPackageVersions)
+                                    {
+                                        if (packageCentralVersion.Key.Equals(topLevelPackage.Name, StringComparison.Ordinal))
```

Comment:
Does [this existing statement](https://github.com/NuGet/NuGet.Client/pull/5350/files#diff-90c9045bb743a57a0e781f15088f964285ff4adf24e006ed4cd3a68d7e520674L754) have a similar bug?  With @martinrrm's changes, the line moved down a bit, but it's still intact.  I didn't scrutinize the code closely, but since we're talking about case insensitivity correctness, the surrounding code is up for discussion.  Not sure why it was case-sensitive to begin with.

Reply:
Yeah, I think that's a bug too.